Test/Tasty/Golden.hs

```diff
@@ -43,6 +47,18 @@ tests will be broken.
 As a last resort, you can strip all @\\r@s from both arguments in your
 comparison function when necessary. But most of the time treating the files
 as binary does the job.
+
+== Linking
+The test harness exectutable into which you import this library
+should be compiled with @-threaded@ if you want to avoid blocking any other
+threads while 'Test.Tasty.Golden' waits for the result of the diff command
+that you specify when using 'goldenVsFileDiff' or 'goldenVsStringDiff'.
+
+== Windows limitations
+When using 'goldenVsFileDiff' or 'goldenVsStringDiff' under Windows the exit
+code from the diff program that you specify will not be captured correctly
+if that program uses @exec@. You may need to use a simpler diff program in
```

Comment:
Maybe you could also add some examples of the programs that use and don't use diff, based on your experience? It's probably hard for an average user to figure out whether a diff program uses exec, so this could be at least a starting point.

Also, the comment below (about why exactly exec is a problem) deserves to be moved here, as it clarifies quite a bit this situation.